modules/programs/alot.nix

```diff
@@ -0,0 +1,145 @@
+# alot config loader is sensitive to leading space !
+{ config, lib, pkgs, ... } @ top:
+
+with lib;
+with import ../lib/dag.nix { inherit lib; };
+
+let
+
+  cfg = config.programs.alot;
+
+  notmuchConfig = account: "$XDG_CONFIG_HOME/notmuch/notmuchrc";
+
+  alot_accounts = config.accounts.email.accounts;
+
+  # https://alot.readthedocs.io/en/latest/configuration/contacts_completion.html
+  contactCompletionStr = let
+      contactCompletion = "notmuch address";
+    in
+    account:
+    ''
+      [[[abook]]]
+      type = shellcommand
+      command = '' + (
+    if contactCompletion == "notmuch address" then ''
+      '${pkgs.bash}/bin/bash -c "${pkgs.notmuch}/bin/notmuch  address --format=json --output=recipients  date:1Y.."'
+    regexp = '\[?{"name": "(?P<name>.*)", "address": "(?P<email>.+)", "name-addr": ".*"}[,\]]?'
+    shellcommand_external_filtering = False
+    '' else if contactCompletion == "notmuch --config ${notmuchConfig account} address simple" then
+    "'${pkgs.notmuch}/bin/notmuch address --format=json date:1Y..'"
+    else "");
+
+  sendMailCommand = account:
+  if account.alot ? sendMailCommand then account.alot.sendMailCommand
+  else if account.msmtp.enable then programs.msmtp.sendCommand account else "";
+
+  accountStr = {name, userName, address, realName, ...} @ account:
+    let
+      showSignature = "none";
+    in
+    ''
+      [[${name}]]
+      address=${address}
+      realname=${realName}
+      sendmail_command = ${sendMailCommand account}
+      signature_as_attachment = ${if showSignature == "attach" then "True" else "False"}
+
+      # contact completion
+      ${contactCompletionStr account}
+      ${if showSignature == "attach" then "gpg_key = {account." else ""}
+    ''
+    ;
+
+configFile = mailAccounts:
+    pkgs.writeText "alot.conf" (''
+# Generated by home-manager
+# see http://alot.readthedocs.io/en/latest/configuration/config_options.html
+hooksfile = "$XDG_CONFIG_HOME/alot/hm_hooks"
+${cfg.extraConfig}
+
+[bindings]
+${cfg.bindings}
+
+[accounts]
+
+${concatStringsSep "\n" (map accountStr mailAccounts)}
+'');
+
+in
+{
+
+  options = {
+    programs.alot = {
+      enable = mkEnableOption "Alot";
+
+      extraHookFile = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Content to append to the hooks file (after home-manager python code)
+        '';
+      };
+
+      contactCompletionCommand = mkOption {
+        default = contactCompletionStr;
+        description = "Can override what is decided in contactCompletion";
```

Comment:
Need some better description that doesn't assume knowledge of the module implementation.